pkg/azdiskschedulerextender/main.go

```diff
@@ -116,13 +116,6 @@ func handleFilterRequest(response http.ResponseWriter, request *http.Request) {
 		return
 	}
 
-	allNodes := args.Nodes.Items
-	if len(allNodes) == 0 {
```

Comment:
Should we make an equivalent check of `args.NodeNames`?

Reply:
I am not sure we need this check in main and respond with BadRequest, we were not doing this for prioritize. I think if nodes are empty, we should just return an empty list?

Reply:
I'm not sure what the expected behavior is if an empty list is passed by kube-scheduler here. Given that we don't generally fail operations on unexpected input and simply return what was given to us, returning an empty list in this case seems reasonable.